test/e2e/tests/test_sagemaker_endpoint_autoscaling.py

```diff
@@ -0,0 +1,200 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the Application Auto Scaling ScalingPolicy API.
+"""
+
+import boto3
+import pytest
+import logging
+from typing import Dict, Tuple
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_autoscaling_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import TestBootstrapResources, get_bootstrap_resources
+from e2e.common.sagemaker_utils import (
+    sagemaker_client,
+    wait_sagemaker_endpoint_status,
+    sagemaker_make_model,
+    sagemaker_make_endpoint_config,
+    sagemaker_make_endpoint,
+)
+
+
+TARGET_RESOURCE_PLURAL = "scalabletargets"
+POLICY_RESOURCE_PLURAL = "scalingpolicies"
+ENDPOINT_CONFIG_RESOURCE_PLURAL = "endpointconfigs"
+MODEL_RESOURCE_PLURAL = "models"
+ENDPOINT_RESOURCE_PLURAL = "endpoints"
+ENDPOINT_STATUS_INSERVICE = "InService"
+
+
+@pytest.fixture(scope="module")
+def name_suffix():
+    return random_suffix_name("sagemaker-endpoint", 32)
+
+
+@pytest.fixture(scope="module")
+def sagemaker_endpoint(name_suffix):
+    model_name = name_suffix + "-model"
+    endpoint_config_name = name_suffix + "-config"
+    endpoint_name = name_suffix
+    resource_id = f"endpoint/{endpoint_name}/variant/variant-1"
+
+    model_input, model_response = sagemaker_make_model(model_name)
+    endpoint_config_input, endpoint_config_response = sagemaker_make_endpoint_config(
+        model_name, endpoint_config_name
+    )
+    endpoint_input, endpoint_response = sagemaker_make_endpoint(
+        endpoint_name, endpoint_config_name
+    )
+
+    wait_sagemaker_endpoint_status(endpoint_name, ENDPOINT_STATUS_INSERVICE)
+
+    yield resource_id
+    sagemaker_client().delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client().delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client().delete_model(ModelName=model_name)
```

Comment:
Can we move this fixture into a `bootstrap` resource? We treat the DynamoDB tables as bootstrapped resources since they're not directly autoscaling CR fixtures. Also moving it into a bootstrap resource will probably help in the future when doing soak tests, so we aren't constantly building and breaking fixtures unnecessarily.